src/detours.cpp

```diff
@@ -1648,6 +1654,9 @@ LONG WINAPI DetourTransactionAbort()
     s_pPendingThreads = NULL;
     s_nPendingThreadId = 0;
 
+    // There is nothing we can do if this fails.
+    HeapUnlock(GetProcessHeap());
```

Comment:
We should really report this if it fails, because "nobody can use the global heap anymore" seems like a pretty big problem

Reply:
Also, since the heap is locked after s_nPendingThreadId is set in DetourTransactionBegin, we should probably unlock the heap before allowing another thread to start a transaction, so before resetting `s_nPendingThreadId ` here